src/main/java/com/redhat/devtools/intellij/tektoncd/tkn/TknCli.java

```diff
@@ -19,20 +19,23 @@
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.openshift.client.OpenShiftClient;
 
-import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.StringReader;
 import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class TknCli implements Tkn {
-    private static final ObjectMapper JSON_MAPPER = new ObjectMapper(new JsonFactory());
+    private static final ObjectMapper TASKRUN_JSON_MAPPER = new ObjectMapper(new JsonFactory());
+    private static final ObjectMapper PIPERUN_JSON_MAPPER = new ObjectMapper(new JsonFactory());
 
     static {
-        SimpleModule module = new SimpleModule();
-        module.addDeserializer(List.class, new TaskRunDeserializer());
-        JSON_MAPPER.registerModule(module);
+        SimpleModule tr_module = new SimpleModule();
+        tr_module.addDeserializer(List.class, new TaskRunDeserializer());
+        TASKRUN_JSON_MAPPER.registerModule(tr_module);
+
+        SimpleModule pr_module = new SimpleModule();
+        pr_module.addDeserializer(List.class, new PipelineRunDeserializer());
+        PIPERUN_JSON_MAPPER.registerModule(pr_module);
```

Comment:
Note sure you need 2 distincts mappers, only one with 2 registers serializers should be ok but approving as I'm not sure

Reply:
By using a single mapper it always uses the last deserializer added so when expanding a node you can end up having an issue like "pipelinerun node cannot be cast to taskrun..."